packages/x-charts/src/hooks/useBarSeries.ts

```diff
@@ -0,0 +1,35 @@
+'use client';
+import { ProcessedSeries } from '../internals/plugins/corePlugins/useChartSeries/useChartSeries.types';
+import { SeriesId } from '../models/seriesType/common';
+import { ChartSeriesDefaultized } from '../models/seriesType/config';
+import { createSeriesSelectorsOfType } from '../internals/createSeriesSelectorOfType';
+
+const selectorSeries = createSeriesSelectorsOfType('bar');
+
+/**
+ * Get access to the internal state of bar series.
+ * The returned object contains:
+ * - series: a mapping from ids to series attributes.
+ * - seriesOrder: the array of series ids.
+ * @returns {{ series: Record<SeriesId, DefaultizedBarSeriesType>; seriesOrder: SeriesId[]; } | undefined}  barSeries
+ */
+export function useBarSeries(): ProcessedSeries['bar'];
+/**
+ * Get access to the internal state of bar series.
+ *
+ * @param {SeriesId} seriesId The id of the series to get.
+ * @returns {ChartSeriesDefaultized<'bar'> | undefined}  barSeries
+ */
+export function useBarSeries(seriesId: SeriesId): ChartSeriesDefaultized<'bar'> | undefined;
+/**
+ * Get access to the internal state of bar series.
+ *
+ * @param {SeriesId[]} seriesIds The ids of the series to get. Order is preserved.
+ * @returns {ChartSeriesDefaultized<'bar'>[] | undefined}  barSeries
+ */
+export function useBarSeries(
+  ...seriesIds: SeriesId[]
+): (ChartSeriesDefaultized<'bar'> | undefined)[];
+export function useBarSeries(...seriesIds: SeriesId[]): any {
+  return selectorSeries(seriesIds);
+}
```

Comment:
Should we divide the logic for `useAllScatterSeries/useScatterSeriesAll` from the rest? The return values are wildly different 🤔

Reply:
I would add it to another PR though 😅

Reply:
I agree. Finding a good name for this hook is the main difficulty. Maybe `useScatterSeriesContext` or `useScatterSeriesState` even if it does not exactly translate what it is